Clarify that minimum viewport scale may change

klausw opened the pull request:
The minimum viewport scale must ensure that the viewport has a nonzero
size, and this may in theory require an adjustment if the session's
baseLayer is changed during a session to use a very small
framebufferScaleFactor.

Change the wording to clarify that it's not necessarily constant for an
entire session.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/klausw/webxr/pull/1134.html" title="Last updated on Sep 30, 2020, 1:22 AM UTC (919e516)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/immersive-web/webxr/1134/132fa32...klausw:919e516.html" title="Last updated on Sep 30, 2020, 1:22 AM UTC (919e516)">Diff</a>

klausw commented:
On second thought, I think it would be easier to enforce the nonzero size
in getViewport, then this change isn't needed and there's less ambiguity
about baseLayer changes.

On Tue, Sep 29, 2020, 15:59 Klaus Weidner <notifications@github.com> wrote:

> The minimum viewport scale must ensure that the viewport has a nonzero
> size, and this may in theory require an adjustment if the session's
> baseLayer is changed during a session to use a very small
> framebufferScaleFactor.
>
> Change the wording to clarify that it's not necessarily constant for an
> entire session.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/immersive-web/webxr/pull/1134
> Commit Summary
>
>    - Clarify that minimum viewport scale may change
>
> File Changes
>
>    - *M* index.bs
>    <https://github.com/immersive-web/webxr/pull/1134/files#diff-ec9cfa5f3f35ec1f84feb2e59686c34d>
>    (2)
>
> Patch Links:
>
>    - https://github.com/immersive-web/webxr/pull/1134.patch
>    - https://github.com/immersive-web/webxr/pull/1134.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/immersive-web/webxr/pull/1134>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKT5XYAR5TRQWXX6SH22FTSIJRD7ANCNFSM4R6MHFDA>
> .
>

cabanier commented:
> On second thought, I think it would be easier to enforce the nonzero size in getViewport, then this change isn't needed and there's less ambiguity about baseLayer changes.

Yes, you can update `getViewPort` to make that adjustment. 
Are there UA that require a certain minimum size for the viewport? If so, you could make the adjustment in that function as well.

klausw commented:
> > On second thought, I think it would be easier to enforce the nonzero size in getViewport, then this change isn't needed and there's less ambiguity about baseLayer changes.
> 
> Yes, you can update `getViewPort` to make that adjustment.

I've moved the logic there.

> Are there UA that require a certain minimum size for the viewport? If so, you could make the adjustment in that function as well.

I'm not aware of a specific UA requirement (beyond ensuring nonzero size), but I added an optional adjustment there where the UA could clamp to a minimum viewport scale.
